### ISSUE TITLE
refsutil does not work on server 2022 with refs version 3.7

### ISSUE THREAD

**Issue by clonea1** [MicrosoftDocs/windowsserverdocs#6950]
refs is in need of update 
ReFS version: 3.7
Error: The volume is an unsupported ReFS version. This utility supports versions up to 3.5. Volume is 3.7.
Error: Command failed.
Error: The volume does not contain a recognized file system. Please make sure that all required file system drivers are loaded and that the volume is not corrupt.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 6a32ad5a-e2ce-a010-2e33-00fd128b25aa
* Version Independent ID: e9c6f167-65fd-6c02-efdc-cfd0143fabe7
* Content: [ReFSUtil](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/refsutil)
* Content Source: [WindowsServerDocs/administration/windows-commands/refsutil.md](https://github.com/MicrosoftDocs/windowsserverdocs/blob/main/WindowsServerDocs/administration/windows-commands/refsutil.md)
* Product: **windows-server**
* Technology: **windows-commands**
* GitHub Login: @laknight5
* Microsoft Alias: **laknight**

**Comment by morgenman** [comment]
I have the same issue, but my version is 3.9. Very frustrating!

**Comment by wbsmolen** [comment]
@clonea1 @morgenman Can you both share what version and SKU of Windows you're running?

@morgenman Presumably you're on an Insider build. I don't think 3.9 is in-market anywhere yet, but the point is valid that refsutil should always be compatible.

**Comment by clonea1** [comment]
Hopefully this is the information you are looking for


OS Name             Microsoft Windows Server 2022 Standard
Version                 10.0.20348 Build 20348

C:\Users\Administrator>refsutil salvage -qs y: r:
Microsoft ReFS Salvage [Version 10.0.11070]
Copyright (c) 2015 Microsoft Corp.

Local time: 2/14/2023 17:48:05

ReFS version: 3.7
Error: The volume is an unsupported ReFS version. This utility supports versions up to 3.5. Volume is 3.7.
Error: Command failed.
Error: The volume does not contain a recognized file system. Please make sure that all required file system drivers are loaded and that the volume is not corrupt.

Run time = 0 seconds.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by wbsmolen** [comment]
It does help - we have this fixed in the latest Windows Server 2022 Datacenter: Azure Edition builds but we're working to get this backported to the in-market Windows Server 2022 SKUs that run on-prem. 

Unfortunately I don't have a timeline for you today. I'll follow up ASAP.

**Comment by clonea1** [comment]
Thanks so far I have only had refs corruption on one of the volumes I used for client backups with dedup running.  The first time was due to storage bus cache drive failure and thankfully the volume remounted after the failed bus cache disk was removed so no data loss.  But the most recent time I could not recover the volume so I purged it instead.  But no prod data as just backups, so I lost historical roll back but restored with a fresh volume backup services.  So far though yeh these refs corruptions circle around dedup volumes for me. And then with out a working util...

Anway good to see it was fixed I also noticed that w11 refs util is a newer version but I couldn't figure out a way to hack and slash that into server 2022.  And no way I would be able to mount my storage spaces virtual disks to client os lol
________________________________

**Comment by wbsmolen** [comment]
@clonea1 try copying refsutil.exe from Win11 and pasting it into C:\Windows\System32 on the affected system. That should work for ReFS 3.7

**Comment by clonea1** [comment]
Wish I had a corrupt drive again to re test, will do it again tonight but I recall having some type of error after that my assumptions at the time where that I was missing some paired dlls or the loke
________________________________

**Comment by wbsmolen** [comment]
> where that I was missing some paired dlls or the loke

If you can repro this, I would love to chat live. Feel free to DM me here on GitHub, or if you have Teams federated with Microsoft's tenant, you can ping me directly at billy@microsoft.com

**Comment by yayatichothe** [comment]
I copied refsutil.exe from Windows 11 Dev to Windows Server 2022 and it worked for me. there is no error of "Error: The volume is an unsupported ReFS version. This utility supports versions up to 3.5. Volume is 3.7."
You will find it under refsutil.exe C:\Windows\System32. 
However, my volume is corrupt and unrecoverable :(

**Comment by clonea1** [comment]
Yup found that it had to be a win 11 22h1.i originally tested on 21h2 and that had failes
________________________________

**Comment by RolloT8p** [comment]
Hi all. I'm in a bit of a pickle with ReFS. Here's a timeline:

-Server hardware failed. ReFS drive in question had dedupe enabled on Server 2022, failure was related to motherboard.
-I plugged drive into a Windows 11 22H2 workstation to verify drive was fine. ReFS updated to 3.9 (jfc Microsoft)
-I plugged the ReFS deduped volume back into replacement server hardware, shows up as RAW (incompatible with Server 2022)

Windows 11 22H2 doesn't recognize deduped blocks. I've tried using tools like UFSExplorer but it fails, cannot read data when trying to copy volume. Using salvage and copying the contents to a new disk still carries the corruption over. 

Any suggestions are greatly appreciated. ReFS is incredibly fragile, this is unfortunate.

**Comment by clonea1** [comment]
Correct the client sku OS does not support dedup feature enable, their used to be some old back door ways to get dedup working on client os but it was fickle ever time a feature update came down.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by robinharwood** [comment]
We hugely appreciate your contribution, feedback, and support to our technical documentation.

We’ve phased out GitHub Issues on this repository as the feedback mechanism for content and have replaced it with a new feedback system. For more information see, [Content User Feedback](https://aka.ms/ContentUserFeedback). As part of this rollout, we are migrating your issue where our team will use your comments to improve the documentation experience. Unfortunately, this does mean you won't be able to access this issue after the migration is complete.

All issues will be migrated by Thursday 29th August 2024.

Once again, thank you for your valuable feedback, we hope you’ll continue to contribute using our new feedback system.
